src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyCultureProvider.cs

```diff
@@ -63,45 +65,31 @@ public void ThrowIfCultureChangeIsUnsupported()
 
     public virtual async ValueTask LoadCurrentCultureResourcesAsync()
     {
-        var culturesToLoad = GetCultures(CultureInfo.CurrentCulture);
-
-        if (culturesToLoad.Count == 0)
+        if (!OperatingSystem.IsBrowser())
         {
-            return;
+            throw new PlatformNotSupportedException("This method is only supported in the browser.");
         }
 
-        // Now that we know the cultures we care about, let WebAssemblyResourceLoader (in JavaScript) load these
-        // assemblies. We effectively want to resovle a Task<byte[][]> but there is no way to express this
-        // using interop. We'll instead do this in two parts:
-        // getSatelliteAssemblies resolves when all satellite assemblies to be loaded in .NET are fetched and available in memory.
-#pragma warning disable CS0618 // Type or member is obsolete
-        var count = (int)await _invoker.InvokeUnmarshalled<string[], object?, object?, Task<object>>(
-            GetSatelliteAssemblies,
-            culturesToLoad.ToArray(),
-            null,
-            null);
-
-        if (count == 0)
+        var culturesToLoad = GetCultures(CultureInfo.CurrentCulture);
+
+        if (culturesToLoad.Length == 0)
         {
             return;
         }
 
-        // readSatelliteAssemblies resolves the assembly bytes
-        var assemblies = _invoker.InvokeUnmarshalled<object?, object?, object?, object[]>(
-            ReadSatelliteAssemblies,
-            null,
-            null,
-            null);
-#pragma warning restore CS0618 // Type or member is obsolete
+        await WebAssemblyCultureProviderInterop.LoadSatelliteAssemblies(culturesToLoad, LoadSatelliteAssembly);
+    }
 
-        for (var i = 0; i < assemblies.Length; i++)
-        {
-            using var stream = new MemoryStream((byte[])assemblies[i]);
-            AssemblyLoadContext.Default.LoadFromStream(stream);
-        }
+    [SupportedOSPlatform("browser")]
+    private void LoadSatelliteAssembly(JSObject wrapper)
+    {
+        byte[] dllBytes = wrapper.GetPropertyAsByteArray("dll")!;
```

Comment:
Just for my education about this JS interop mechanism, could you clarify why we're passing a `JSObject` here and then calling `GetPropertyAsByteArray`? Would it also be possible just to have a `byte[]` as the method parameter in the first place, or do we not support marshalling `Action<byte[]>`?

Reply:
Also nit:

```suggestion
        var dllBytes = wrapper.GetPropertyAsByteArray("dll")!;
```

Reply:
Because we don't have support for passing `Task<byte[]>` in Net7. That's something I would love to improve.